### PR TITLE
Require files from the .cache directory so user can run es/commonjs

### PR DESCRIPTION
**Pull request by abhiaiyer91**
Getting errors running functions written with default exports

```
 ERROR 

/Users/abhiaiyer/gatsby/gatsby-functions/src/functions/dog.ts:3
export default async (req, res) => {
^^^^^^

SyntaxError: Unexpected token 'export'
```

So moved the require to the compiled dir.

### CODE REVIEW

File: packages/gatsby-plugin-functions/src/gatsby-node.ts [hunk 1]
```diff
@@ -67,51 +58,57 @@ export async function onPreBootstrap(
...
-        })
-      )
-    })
-  )
+        )
+      })
+    )
+  } catch (e) {
+    reporter.error(`Failed to compile Gatsby Functions.`)
+    reporter.panicOnBuild(e)
```
Comment (wardpeet) [inline packages/gatsby-plugin-functions/src/gatsby-node.ts:110]: Any reason why you only panic on build and not in develop?
```suggestion
    activity.panicOnBuild(`Failed to compile Gatsby Functions.`, e)
```

File: packages/gatsby-plugin-functions/src/gatsby-node.ts [hunk 2]
```diff
@@ -129,26 +126,43 @@ export async function onCreateDevServer(
...
+      activity.start()
+      const compiledFunctionsDir = path.join(
+        process.cwd(),
+        `.cache`,
+        `functions`
+      )
+      const funcNameToJs = (knownFunctions.get(functionName) as string).replace(
+        `ts`,
+        `js`
+      )
```
Comment (wardpeet) [inline packages/gatsby-plugin-functions/src/gatsby-node.ts:154]: We should get this information from webpack instead. It's fine for now :)

File: packages/gatsby-plugin-functions/src/gatsby-node.ts [hunk 3]
```diff
@@ -129,26 +126,43 @@ export async function onCreateDevServer(
...
     const { functionName } = req.params
 
     if (knownFunctions.has(functionName)) {
-      const fn = require(path.join(
-        functionsDirectory,
-        knownFunctions.get(functionName) as string
-      ))
+      const activity = reporter.activityTimer(
+        `Executing function ${functionName}`
+      )
```
Comment (wardpeet) [inline packages/gatsby-plugin-functions/src/gatsby-node.ts:144]: Reporter will be unknown inside a cloud-run container, we'll need to replace it with a cloud alternative. For now, I suggest using `console.log`

**Review by wardpeet — COMMENTED** [review]
Added some comments :)

File: packages/gatsby-plugin-functions/src/gatsby-node.ts (hunk 3 above)
Comment (julienp) [reply packages/gatsby-plugin-functions/src/gatsby-node.ts:144]: This is the the handler in gatsby develop, in cloud-run we have a different handler to call the function.

File: packages/gatsby-plugin-functions/src/gatsby-node.ts (hunk 1 above)
Comment (abhiaiyer91) [reply packages/gatsby-plugin-functions/src/gatsby-node.ts:110]: What should it be instead if you want to panic in both?

File: packages/gatsby-plugin-functions/src/gatsby-node.ts (hunk 2 above)
Comment (abhiaiyer91) [reply packages/gatsby-plugin-functions/src/gatsby-node.ts:154]: how?

File: packages/gatsby-plugin-functions/src/gatsby-node.ts (hunk 1 above)
Comment (wardpeet) [reply packages/gatsby-plugin-functions/src/gatsby-node.ts:110]: just panic :)

**Review by sidharthachatterjee — APPROVED** [review]
Thank you!

**Pull request merged**